assets/src/edit-story/utils/getCaretCharacterOffsetWithin.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * This function finds the character offset within a certain element.
+ *
+ * For instance, given the following node with `_` marking the current selection start offset:
+ *
+ * ```
+ * <p>Corgies are the <em>b_est</em>!</p>
+ * ```
+ *
+ * This function would return 17 (`'Corgies are the b'.length`).
+ *
+ * If optional coordinates are given, the point under the coordinates will be used,
+ * otherwise the current on-page selection will be used.
+ *
+ * This function includes some cross-browser optimization for older browsers even
+ * though they aren't really supported by the editor at large (IE).
+ *
+ * @param {Node} element    DOM node to find current selection within.
+ * @param {number} clientX  Optional x coordinate of click.
+ * @param {number} clientY  Optional y coordinate of click.
+ * @return {number} Current selection start offset as seen in `element` or 0 if not found.
+ */
+function getCaretCharacterOffsetWithin( element, clientX, clientY ) {
+	const doc = element.ownerDocument || element.document;
+	const win = doc.defaultView || doc.parentWindow;
+	let sel;
+	if ( typeof win.getSelection !== 'undefined' ) {
+		sel = win.getSelection();
+		if ( sel.rangeCount > 0 ) {
+			let range = win.getSelection().getRangeAt( 0 );
+			if ( clientX && clientY ) {
+				if ( doc.caretPositionFromPoint ) {
+					range = document.caretPositionFromPoint( clientX, clientY );
+				} else if ( doc.caretRangeFromPoint ) {
+					range = document.caretRangeFromPoint( clientX, clientY );
+				}
```

Comment:
So, both of these methods are non-standard, but cover most browsers currently.

[The former, `caretPositionFromPoint`, is drafted](https://drafts.csswg.org/cssom-view/#dom-document-caretpositionfrompoint), but [not implemented in Chrome or Edge](https://developer.mozilla.org/en-US/docs/Web/API/DocumentOrShadowRoot/caretPositionFromPoint#Browser_compatibility) so far.